src/node/rpc/frontend.h

```diff
@@ -352,6 +354,48 @@ namespace ccf
         }
       }
 
+      std::optional<Jwt> jwt;
+      if (endpoint->properties.require_jwt_authentication)
+      {
+        auto headers = ctx->get_request_headers();
+        std::string error_reason;
+        auto token = http::JwtVerifier::extract_token(headers, error_reason);
+        auto keys_view = tx.get_view<JwtPublicSigningKeys>(
+          ccf::Tables::JWT_PUBLIC_SIGNING_KEYS);
+        auto key_issuer_view = tx.get_view<JwtPublicSigningKeyIssuer>(
+          ccf::Tables::JWT_PUBLIC_SIGNING_KEY_ISSUER);
+        std::string key_issuer;
+        if (token.has_value())
+        {
+          auto key_id = token.value().header_typed.kid;
+          auto token_key = keys_view->get(key_id);
+          if (!token_key.has_value())
+          {
+            error_reason = "JWT signing key not found";
+          }
+          else if (!http::JwtVerifier::validate_token_signature(
+                     token.value(), token_key.value()))
+          {
+            error_reason = "JWT signature is invalid";
+          }
+          else
+          {
+            key_issuer = key_issuer_view->get(key_id).value();
+          }
+        }
+        if (!error_reason.empty())
+        {
+          set_response_unauthorized(
```

Comment:
This sets a header telling the client to use `Signature` auth. In this case we should instead indicate that this requires `Bearer` auth.